File: Data Structures & Algorithms/maximum-subarray/submission-1.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int currSum = 0;
        int maxSum = nums[0];

        for (int num : nums) {
            currSum = max(num, currSum + num);

            if (currSum > maxSum) {
                maxSum = currSum;
            }
        }

        return maxSum;
    }
};

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-1.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        l = [0] * len(nums)
        maxS = 1

        for i in range(len(nums)):
            l[i] = 1
            for j in range(i):
                if nums[j] < nums[i]:
                    l[i] = max(l[i], 1 + l[j])
                    maxS = max(maxS, l[i])
        
        return maxS